### ISSUE TITLE
[package] ninja/1.11.1: even after builing it, I cannot re-use it in meson for example

### ISSUE THREAD

**Issue by hpe-ykoehler** [conan-io/conan-center-index#23274]
### Description

I am trying to build all the recipe from conan-center-index (my own fork) that I need.
I need meson, so I need to build ninja.

I build ninja using conan source/conan build -nr (I do not want any remote to be used) and then conan export to export to my local cache.

I then build meson, and I get the error that the ninja package cannot be found.

### Package and Environment Details

* Package Name/Version: **ninja/1.11.1**
* Operating System+version: **Linux Ubuntu 20.04**
* Compiler+version: **GCC 9**
* Docker image:
* Conan version: **conan 2.2.1**
* Python version: **Python 3.7.4**

### Conan profile

Profile host:
[settings]
arch=x86_64
build_type=RelWithDebInfo
compiler=gcc
compiler.cppstd=gnu14
compiler.libc=glibc
compiler.libc.version=2.31
compiler.libcxx=libstdc++11
compiler.version=9
os=Linux
os.platform=native
os.version=5.15
[options]
nanopb/*:shared=True
protobuf/*:with-zlib=False
protobuf/3.21.9:with_python=True
[conf]
tools.cmake.cmake_layout:build_folder_vars=['settings.os.platform']
tools.build:compiler_executables={'c': '/usr/bin/gcc'}

Profile build:
[settings]
arch=x86_64
build_type=RelWithDebInfo
compiler=gcc
compiler.cppstd=gnu14
compiler.libc=glibc
compiler.libc.version=2.31
compiler.libcxx=libstdc++11
compiler.version=9
os=Linux
os.platform=native
os.version=5.15
[options]
nanopb/*:shared=True
protobuf/*:with-zlib=False
protobuf/3.21.9:with_python=True
[conf]
tools.cmake.cmake_layout:build_folder_vars=['settings.os.platform']
tools.build:compiler_executables={'c': '/usr/bin/gcc'}

### Steps to reproduce

rm -rf ~/.conan2/p

conan source recipes/ninja/all --name ninja --version 1.11.1
conan build recipes/ninja/all --name ninja --version 1.11.1 -nr
conan export recipes/ninja/all --name ninja --version 1.11.1

conan source recipes/meson/all --name meson --version 1.2.2
conan build recipes/meson/all --name ninja --version 1.2.2 -nr

### Logs

<details><summary>Click to expand log</summary>

+ conan build all -pr:h default -pr:b default -s:h build_type=RelWithDebInfo -s:b build_type=RelWithDebInfo --name meson --version 1.2.2 -nr

======== Input profiles ========
Profile host:
[settings]
arch=x86_64
build_type=RelWithDebInfo
compiler=gcc
compiler.cppstd=gnu14
compiler.libc=glibc
compiler.libc.version=2.31
compiler.libcxx=libstdc++11
compiler.version=9
os=Linux
os.platform=native
os.version=5.15
[options]
nanopb/*:shared=True
protobuf/*:with-zlib=False
protobuf/3.21.9:with_python=True
[conf]
tools.cmake.cmake_layout:build_folder_vars=['settings.os.platform']
tools.build:compiler_executables={'c': '/usr/bin/gcc'}

Profile build:
[settings]
arch=x86_64
build_type=RelWithDebInfo
compiler=gcc
compiler.cppstd=gnu14
compiler.libc=glibc
compiler.libc.version=2.31
compiler.libcxx=libstdc++11
compiler.version=9
os=Linux
os.platform=native
os.version=5.15
[options]
nanopb/*:shared=True
protobuf/*:with-zlib=False
protobuf/3.21.9:with_python=True
[conf]
tools.cmake.cmake_layout:build_folder_vars=['settings.os.platform']
tools.build:compiler_executables={'c': '/usr/bin/gcc'}

======== Computing dependency graph ========
Graph root
    conanfile.py (meson/1.2.2): /work/conan-center-index/recipes/meson/all/conanfile.py
Requirements
    ninja/1.11.1#de85db1490cc53d3353a536dadf1db50 - Cache

======== Computing necessary packages ========
ninja/1.11.1: Compatible package ID 19546e8105b49315e449ada1d8eb6678079e7133 equal to the default package ID: Skipping it.
Requirements
    ninja/1.11.1#de85db1490cc53d3353a536dadf1db50:19546e8105b49315e449ada1d8eb6678079e7133 - Missing

======== Installing packages ========

======== Installing packages ========
ERROR: Missing binary: ninja/1.11.1:19546e8105b49315e449ada1d8eb6678079e7133

ninja/1.11.1: WARN: Can't find a 'ninja/1.11.1' package binary '19546e8105b49315e449ada1d8eb6678079e7133' for the configuration:
[settings]
arch=x86_64
build_type=RelWithDebInfo
os=Linux
os.platform=native
os.version=5.15

ERROR: Missing prebuilt package for 'ninja/1.11.1'
Check the available packages using 'conan list ninja/1.11.1:* -r=remote'
or try to build locally from sources using the '--build=ninja/1.11.1' argument

More Info at 'https://docs.conan.io/2/knowledge/faq.html#error-missing-prebuilt-package'

</details>

**Comment by hpe-ykoehler** [comment]
I think it may be my fault, the export seems to only export the recipe, not the binary to the cache, I may be missing an export-pkg or something like that.

**Comment by jcar87** [comment]
Hi @hpe-ykoehler  - 

with a local copy of the ninja and meson recipes, please try:

```
conan create recipes/ninja/all --version=1.11.1 --profile:host [HOST_PROFILE] --profile:build [BUILD_PROFILE]    --build-require --no-remote
```

where you use your host and build profiles. UNless you know exactly which platform ninja will run _on_, in this case you can skip the `--build-require ` flag

Same would go for meson!

**Comment by hpe-ykoehler** [comment]
yeah but my goal is to build locally and export, so I do not want to use "create", in any case, I have solved this by replacing export with export-pkg.

**Comment by hpe-ykoehler** [comment]
not a real issue, was not using conan properly.
